### ISSUE TITLE
Elastalert rule for hits over time

### ISSUE THREAD

**Issue by PiperMp3** [Yelp/elastalert#2356]
Hello,

I got 7-8 rules running, frequency and cardinality and everything is great.

I'm looking for a rule setup that would accomplish the following;
When ElastAlert sees ValueA in FieldA - Remember the timestamp, Ie "I can see one hit at 11:10 on the 24/07/2019" which is when I'm writing this post.

I then want ElastAlert to keep checking, to see if it can get the same HIT, after a certain time.
For this example, lets say 3 minutes.

Ie, 1st Hit at 11:10 - no more hits by 11:13 - no alerts
1st hit at 11:10 - another hit at 11:13 (3 minutes) - ok those events have a timestamp difference of 3 minutes - Alert
1.st hit at 11:10 - do nothing, nothing happens at 11:11, 11:12, 11:13 - reset to 0 hits.
Another hit after 4 minutes (11:14) - It's at 1 hit, First hit known as 11:14.

Basically, there are certain events that we only wants alert for, if they "persist" over a timespan.
We cannot use Frequency or Thresholds, as the number of events are not important.

The only important part is if the event we hit, continues over a prolonged period of time - in this example 3 minutes. We dont care if its 2 hits, or 400.

Anyone had the need for this? Any examples or advice? 

Cheers

**Comment by Qmando** [comment]
I think you can accomplish this using cardinality type.

```
query_key: fieldA
cardinality_field: _id
max_cardinality 1
timeframe:
  minutes: 3
```

This way it will only alert if there two different _ids for the same value of fieldA within 3 minutes of each other.
